From c02de4356749748d2ded3f21f5f63bb92ca7b29c Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sun, 15 Sep 2024 16:07:01 +0300
Subject: [PATCH] [FSW-6580] fs1: add boot info tag

With this patch, added string to identify boot
info files written using lfs.
There is correction in forming lfs file name from
file tag and file type.

Change-Id: I4874af240d8cbc51405e37e005f3d31e0231991e
---

diff --git a/arch/fs1/arc/dev_boot_info.h b/arch/fs1/arc/dev_boot_info.h
index 186feed..027448e 100644
--- a/arch/fs1/arc/dev_boot_info.h
+++ b/arch/fs1/arc/dev_boot_info.h
@@ -9,6 +9,8 @@
 #include <soc.h>
 #include <boot_rom.h>
 
+#define LFS_TAG_BOOT_INFO		"boot_info_"
+
 #define OFFSET_32KB		(32 * 1024)
 #define CONFIG_BOOTINFO_ADDR	(HL_FW_MEM_BASE + SCRATCHPAD_SRAM_OFFSET)
 #define CONFIG_BOOTINFO_SIZE	OFFSET_32KB
diff --git a/common/littlefs/lfs_wrapper.c b/common/littlefs/lfs_wrapper.c
index 4181819..cf820c3 100644
--- a/common/littlefs/lfs_wrapper.c
+++ b/common/littlefs/lfs_wrapper.c
@@ -231,6 +231,9 @@
 		file_name[i] = file_tag[i];
 	}
 
+	if (i > 0)
+		i -= 1;
+
 	for (j = 0; file_num_str[j] && (i < LFS_NAME_MAX); j++, i++) {
 		file_name[i] = file_num_str[j];
 	}
